torchgeo/datasets/geo.py

```diff
@@ -375,6 +377,15 @@ def __init__(
                 f"No {self.__class__.__name__} data was found in '{root}'"
             )
 
+        if bands and self.all_bands:
+            band_indexes = [self.all_bands.index(i) + 1 for i in bands]
+            self.bands = bands
+            assert len(band_indexes) == len(self.bands)
+        else:
```

Comment:
Should we raise an error `elif bands`? If someone asks for specific bands but doesn't define `all_bands` then that could lead to unexpected behavior.

Reply:
What about making `all_bands` an abstract method? Wouldn't it be sensible to ask users to define the bands of the datasets they create?

Reply:
The problem with that is datasets that only have a single band. Especially for label datasets I don't think that makes sense.